test/functional/test_framework/test_framework.py

```diff
@@ -108,6 +108,22 @@ def main(self):
             os.makedirs(self.options.tmpdir, exist_ok=False)
         else:
             self.options.tmpdir = tempfile.mkdtemp(prefix="test")
+
+        exit_code = TEST_EXIT_PASSED
+        for variant in self.variants:
+            self.variant = variant
+            self.tmpdir = self.options.tmpdir
+            if variant:
+                self.tmpdir = os.path.join(self.options.tmpdir, variant)
+                os.mkdir(self.tmpdir)
+            success = self.run_variant()
+            if success == TestStatus.SKIPPED and exit_code == TEST_EXIT_PASSED:
```

Comment:
Skipping a test for different variant will result in `FAILED`, no?

Reply:
Will fix.